tests/Rules/UnusedPublicPropertyRule/UnusedPublicPropertyRuleTest.php

```diff
@@ -109,6 +109,10 @@ public static function provideData(): Iterator
             ],
             [],
         ];
+
+        yield [[__DIR__ . '/Fixture/UsedInUnionA.php', __DIR__ . '/Fixture/UsedInUnionB.php', __DIR__ . '/Source/UsedInUnion.php'], []];
+        yield [[__DIR__ . '/Fixture/UsedInUnionA.php', __DIR__ . '/Fixture/UsedInUnionB.php', __DIR__ . '/Source/UsedInUnionPhpdoc.php'], []];
```

Comment:
btw: IMO as someone used to the 1-file test-cases from the phpstan-src repo, this separation of classes over several files makes creating unit tests pretty hard and unnecessary complex

Reply:
What do you suggest?

Reply:
I would put all classes for a single test-case into a single file (with a unique namespace)

Reply:
I see. For the `/Source` it made mess as people used same-named classes: )

Reply:
for the PR we have here, that would mean the test-case file would look like:

```
namespace Bug119;

class UsedInUnionA {
    public float $amount;
}

class UsedInUnionB {
    public float $amount;
}

function doFooBar(UsedInUnionA|UsedInUnionB $aOrB): void {
    echo $aOrB->amount;
}

```

Reply:
> same-named classes: )

therefore a dedicated namespace for each test.

Reply:
in phpstan-src there is a CI check to make sure class-names are unique - shipmonk/name-collision-detector

Reply:
> therefore a dedicated namespace for each test.

That's the place people failed :) we had dozens such cases in Rector, often easy to miss in IDE.